canal/circuit.py

```diff
@@ -848,7 +848,9 @@ def __init__(self, tiles: Dict[int, Tile],
         self.__add_tile_id()
         # add ports
         self.add_ports(stall=magma.In(magma.Bits[stall_signal_width]),
-                       reset=magma.In(magma.AsyncReset))
+                       reset=magma.In(magma.AsyncReset),
+                       clk=magma.In(magma.Clock),
```

Comment:
all tiles have stall, reset, and clk ports right?

Reply:
Not necessarily. In Jade we have empty tiles that doesn't contain anything, since we have IOs on four sides. I think there is code somewhere that pops these ports. Just to make sure that these ports are popped correctly.

Reply:
Sure, that code is in the garnet repo I believe. Is that the appropriate place for it?